hy/compiler.py

```diff
@@ -2521,52 +2156,33 @@ def compile_symbol(self, symbol):
             glob = HySymbol(glob).replace(symbol)
             ret = self.compile_symbol(glob)
 
-            ret = ast.Attribute(
-                lineno=symbol.start_line,
-                col_offset=symbol.start_column,
+            return asty.Attribute(
+                symbol,
                 value=ret,
                 attr=ast_str(local),
-                ctx=ast.Load()
-            )
-            return ret
+                ctx=ast.Load())
 
         if symbol in _stdlib:
             self.imports[_stdlib[symbol]].add(symbol)
 
-        return ast.Name(id=ast_str(symbol),
-                        arg=ast_str(symbol),
-                        ctx=ast.Load(),
-                        lineno=symbol.start_line,
-                        col_offset=symbol.start_column)
-
-    @builds(HyString)
-    def compile_string(self, string):
-        return ast.Str(s=str_type(string),
-                       lineno=string.start_line,
-                       col_offset=string.start_column)
-
-    @builds(HyBytes)
-    def compile_bytes(self, bytestring):
-        f = ast.Bytes if PY3 else ast.Str
-        return f(s=bytes_type(bytestring),
-                 lineno=bytestring.start_line,
-                 col_offset=bytestring.start_column)
-
-    @builds(HyKeyword)
-    def compile_keyword(self, keyword):
-        return ast.Str(s=str_type(keyword),
-                       lineno=keyword.start_line,
-                       col_offset=keyword.start_column)
+        return asty.Name(symbol, id=ast_str(symbol), ctx=ast.Load())
+
+    @builds(HyString, HyKeyword, HyBytes)
+    def compile_string(self, string, building):
+        node = asty.Bytes if PY3 and building is HyBytes and PY3 else asty.Str
```

Comment:
`PY3 and building is HyBytes and PY3` checking it twice?

Reply:
Oops, I must've been in a Christmasy mood ("He's makin' a list, he's checkin' it twice…"). Fixed.